src/test/java/bot/utils/NameProcessorTest.java

```diff
@@ -0,0 +1,304 @@
+package bot.utils;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class NameProcessorTest {
+
+    private String input;
+    private List<String> expected;
+    private List<String> output;
+
+    private void runTest() {
+        output = NameProcessor.generateAcceptedNames(input);
+        Collections.sort(expected);
+        Collections.sort(output);
+        assertEquals(output.containsAll(expected), expected.containsAll(output));
```

Comment:
If you check this with `containsAll`, I don't think you have to sort the collections beforehand. Also, you can just check:
`assertThat(expected, contains("foo", "bar"));` in every test instead of invoking `runTest` every time, or just:
`assertTrue(result.containsAll(expected));`
Or, even better:
`Assert.assertThat(output,
                IsIterableContainingInAnyOrder.containsInAnyOrder(expected.toArray()));`
to get the most information about why a test is failing

Reply:
siryessir!